DataFormats/L1TParticleFlow/interface/PFJet.h

```diff
@@ -38,13 +41,22 @@ namespace l1t {
     edm::Ptr<l1t::PFCandidate> daughterPtr(size_type i) const { return constituents_[i]; }
 
     // Get and set the encodedJet_ bits. The Jet is encoded in 128 bits as a 2-element array of uint64_t
-    std::array<uint64_t, 2> encodedJet() { return encodedJet_; }
-    void setEncodedJet(std::array<uint64_t, 2> jet) { encodedJet_ = jet; }
+    // We store encodings both for Correlator internal usage and for Global Trigger
+    enum HWEncoding { CT, GT };
+    std::array<uint64_t, 2> encodedJet(const HWEncoding encoding = HWEncoding::GT) { return encodedJet_[encoding]; }
+    void setEncodedJet(const HWEncoding encoding, const std::array<uint64_t, 2> jet) { encodedJet_[encoding] = jet; }
+
+    // Accessors to HW objects with ap_* types from encoded words
+    l1gt::Jet getHWJetGT(){ return l1gt::Jet::unpack(encodedJet(HWEncoding::GT)); }
+    l1ct::Jet getHWJetCT(){ return l1ct::Jet::unpack(encodedJet(HWEncoding::CT)); }
 
   private:
     float rawPt_;
     Constituents constituents_;
-    std::array<uint64_t, 2> encodedJet_ = {{0, 0}};
+    std::unordered_map<HWEncoding, std::array<uint64_t, 2>> encodedJet_ = {
+      {HWEncoding::CT, {{0, 0}}},
+      {HWEncoding::GT, {{0, 0}}}
+    };
```

Comment:
an unordered map just to hold 2 elements is an big overhead, maybe you could just have an array, and use `static_cast<int>` to downcast the enum to the index in the array.